recipes/icu/all/conanfile.py

```diff
@@ -127,16 +125,14 @@ def generate(self):
             base_path = unix_path(self, self.dependencies.build["icu"].package_folder)
             tc.configure_args.append(f"--with-cross-build={base_path}")
             if self.settings.os in ["iOS", "tvOS", "watchOS"]:
-                gnu_triplet = get_gnu_triplet("Macos", str(self.settings.arch))
-                tc.configure_args.append(f"--host={gnu_triplet}")
-            elif is_msvc(self):
-                # ICU doesn't like GNU triplet of conan for msvc (see https://github.com/conan-io/conan/issues/12546)
-                host = get_gnu_triplet(str(self.settings.os), str(self.settings.arch), "gcc")
-                build = get_gnu_triplet(str(self._settings_build.os), str(self._settings_build.arch), "gcc")
-                tc.configure_args.extend([
-                    f"--host={host}",
-                    f"--build={build}",
-                ])
+                # ICU build scripts interpret all Apple platforms as 'darwin'.
+                # Since this can coincide with the `build` triple, we need to tweak
+                # the build triple to avoid the collision and ensure the scripts
+                # know we are cross-building.
+                host_triplet = f"{str(self.settings.arch)}-apple-darwin"
+                build_triplet = f"{str(self._settings_build.arch)}-apple"
```

Comment:
Are they valid triplet? I mean if arch is armv8, triplet should be aarch64-apple not armv8-apple.

Reply:
I've tested and icu doesn't seem to complain.

Reply:
Indeed - in a `config.sub` this is likely to be appear like:
```
| arm | arm[lb]e | arme[lb] | armv* \
```

so it will work _as long as_ the build scripts don't have further conditionals on the specific arch part of the triple. ICU doesn't seem too bothered. Even when building natively on apple silicon, the arch is autodetected as `arm` (and not `aarch64` like a more modern `config.sub` would do).